services/spar/src/Spar/Error.hs

```diff
@@ -87,6 +99,12 @@ waiToServant waierr@(Wai.Error status label _) = ServantErr
   , errHeaders      = []
   }
 
+sparToWaiErrorIO :: Log.Logger -> SparError -> MonadIO m => m (Either ServantErr Wai.Error)
+sparToWaiErrorIO logger err = do
+  let errPossiblyWai = sparToWaiError err
+  liftIO $ Wai.logError logger Nothing (either servantToWaiError id $ errPossiblyWai)
```

Comment:
The names of these functions should be cleaned up because `sparToWaiError` Doesn't  convert a spar error to a wai error; it converts it to a wai OR a servant error. The function you've written here DOES convert any spar error to a Wai error, so it should have the name `sparErrorToWaiError`; wheras the one which emits an either should be re-named.

Reply:
What do you mean by "the function you've written"?  I don't see a function that converts any spar error to a Wai error.  Or do you mean "the name of the function you've written *suggests* that..."?

Do you have an idea what to call this function?  I'll try `sparToWaiError`.